source/BeamCalReco/src/BeamCalBkgPregen.cpp

```diff
@@ -67,7 +68,9 @@ void BeamCalBkgPregen::init(vector<string> &bg_files, const int n_bx)
   m_backgroundBX = new TChain("bcTree");
 
   //mix up the files, because the random numbers are ordered to avoid repeating
-  std::random_shuffle(bg_files.begin(), bg_files.end());
+  std::random_device seed_gen;
```

Comment:
Isn't it creating a random seed generator with always the same seed ?

Reply:
random_device  is a "non-deterministic" generator, so it should be different every time it is called, if a source of randomness is available.
https://en.cppreference.com/w/cpp/numeric/random/random_device
This is part of the initialize, so the EventSeeder is not available, or is it?

Reply:
The `ProcessorEventSeeder` was meant to be used in the `processEvent()` method, however one could do:
```cpp
Global::EVENTSEEDER->registerProcessor(this);
int seed = Global::EVENTSEEDER->getSeed(this);
```
this then gives you:
```
srand( Global::parameters->getIntVal("RandomSeed" ) );
rand() ; 
```
i.e. the same random seed for every processor...

Reply:
In #64 I created an object based on TRandom3 (which is already in use in the class), to replace random_device. 
Mostly here: https://github.com/FCALSW/FCalClusterer/pull/64/files#diff-596bee12caf71192a40b6a7d45b5ae15